common/src/main/java/com/wynntils/features/debug/ContentBookDumpFeature.java

```diff
@@ -0,0 +1,143 @@
+import com.wynntils.core.keybinds.KeyBind;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.models.activities.type.ActivityDifficulty;
+import com.wynntils.models.activities.type.ActivityInfo;
+import com.wynntils.models.activities.type.ActivityLength;
+import com.wynntils.models.activities.type.ActivityRequirements;
+import com.wynntils.models.activities.type.ActivityType;
+import com.wynntils.models.profession.type.ProfessionType;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.mc.StyledTextUtils;
+import com.wynntils.utils.mc.type.Location;
+import com.wynntils.utils.type.Pair;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import net.minecraft.network.chat.Component;
+import org.lwjgl.glfw.GLFW;
+
+@StartDisabled
+@ConfigCategory(Category.DEBUG)
+public class ContentBookDumpFeature extends Feature {
+    private static final File SAVE_FOLDER = WynntilsMod.getModStorageDir("debug");
+
+    @RegisterKeyBind
+    private final KeyBind dumpContentBook =
+            new KeyBind("Dump Content Book", GLFW.GLFW_KEY_UNKNOWN, true, this::dumpContentBook);
+
+    private List<DumpableActivityInfo> currentDump = List.of();
+
+    private void dumpContentBook() {
+        currentDump = new ArrayList<>();
+
+        Models.Activity.scanContentBook(ActivityType.ALL, (activityInfos, progress) -> {
+            currentDump.addAll(activityInfos.stream()
+                    .map(DumpableActivityInfo::fromActivityInfo)
+                    .toList());
+
+            saveToDisk();
+
+            // Free up memory
+            currentDump = List.of();
+        });
+    }
+
+    private void saveToDisk() {
+        // Save the dump to a file
+        Map<ActivityType, List<DumpableActivityInfo>> mappedActivities = getMappedDumpedActivities();
+
+        JsonElement element = Managers.Json.GSON.toJsonTree(mappedActivities);
+
+        String fileName = "content_book_dump.json";
+        File jsonFile = new File(SAVE_FOLDER, fileName);
+        Managers.Json.savePreciousJson(jsonFile, element.getAsJsonObject());
+
+        McUtils.sendMessageToClient(Component.literal("Saved content book dump to " + jsonFile.getAbsolutePath()));
+    }
+
+    // This method creates a consisnent mapping and ordering for the dumped activities
+    private Map<ActivityType, List<DumpableActivityInfo>> getMappedDumpedActivities() {
+        Map<ActivityType, List<DumpableActivityInfo>> mappedDumpedActivities = new EnumMap<>(ActivityType.class);
+        for (ActivityType value : ActivityType.values()) {
+            mappedDumpedActivities.put(value, new ArrayList<>());
+        }
+
+        for (DumpableActivityInfo activityInfo : currentDump) {
+            ActivityType type = activityInfo.type();
+
+            // Merge STORYLINE_QUEST and QUEST
+            if (type == ActivityType.STORYLINE_QUEST) {
```

Comment:
I think we can keep this. I did not notice when looking at the json dump that it was missing. This is a distinction Wynn makes, there is no reason to throw away information at this stage. We can merge this later on, if we want.

Reply:
We do not throw it away. The type is still there, we just group it in the dump.

Reply:
Aha, I see.